src/coreclr/src/jit/emitxarch.cpp

```diff
@@ -3952,7 +3952,7 @@ void emitter::emitIns_I(instruction ins, emitAttr attr, cnsval_ssize_t val)
 {
     UNATIVE_OFFSET sz;
     instrDesc*     id;
-    bool           valInByte = ((signed char)val == val);
+    bool           valInByte = ((signed char)val == (target_ssize_t)val);
```

Comment:
It looks like you can change signatures of this function to use `cnsval_ssize_t` because there are only 2 users:
- `emitter::instrDesc* emitter::emitNewInstrSC(emitAttr attr, cnsval_ssize_t cns)` - already `cnsval_ssize_t ` ;

- `void emitAdjustStackDepth(instruction ins, ssize_t val)` should be changed to `cnsval_ssize_t` as well.